### PR TITLE
feat(rslint_parser): Parse JSX Fragments

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

Implements parsing of JSX Fragments `<>a</>` 

I decided to rename `JsxAnyElement` to `JsxAnyTag` so that we don't need a new expression node for fragments. Tag also seems to be the agreed term in React:

* Error message if you have multiple root nodes: `/App.js: Adjacent JSX elements must be wrapped in an enclosing tag. Did you want a JSX fragment <>...</>? (6:4)`
* [Documentation](https://beta.reactjs.org/learn/writing-markup-with-jsx#1-return-a-single-root-element) referring to Tag when explaining fragments.

I used this PR as an opportunity to restructure the folder structure for the JSX nodes so that `tag` and `attribute` now have their own folder. 

## Test Plan

Added new parser tests

### CODE REVIEW

File: crates/rslint_parser/src/syntax/jsx/mod.rs [hunk 1]
```diff
@@ -110,58 +110,88 @@ fn parse_jsx_expression(p: &mut CheckpointedParser<'_, '_>) -> ParsedSyntax {
...
+            return Absent;
+        }
+
+        p.bump_with_context(T![>], LexContext::JsxChild);
+
+        parse_children(p);
+
+        if !p.expect(T![<]) || !p.expect(T![/]) || !p.expect(T![>]) {
+            m.abandon(p);
+            return Absent;
```
Comment (ematipico) [inline crates/rslint_parser/src/syntax/jsx/mod.rs:185]: Is it safe `return Absent`? How do we know that `parse_children(p)` didn't advance the parser? We should have a comment explaining why it's safe

File: xtask/codegen/js.ungram [hunk 2]
```diff
@@ -2178,13 +2179,22 @@ JsxClosingElement =
 	name: JsxAnyElementName
 	'>'
 
+// <>content</>
+JsxFragment =
+	opening_l_angle_token: '<'
+	opening_r_angle_token: '>'
+	children: JsxTextLiteral?
```
Comment (ematipico) [inline xtask/codegen/js.ungram:2186]: I don't think this is correct. Fragments can have everything inside, also other jsx tags:

```jsx
<>
	<Header />
	<Body />
	<Footer />
</>
```

This is valid syntax.

File: crates/rslint_parser/src/syntax/jsx/mod.rs [hunk 3]
```diff
@@ -110,58 +110,88 @@ fn parse_jsx_expression(p: &mut CheckpointedParser<'_, '_>) -> ParsedSyntax {
...
+// test jsx fragments
+// <>abcd</>;
+// <>   whitespace
+// </>;
+// <
+//   /*comment */
+//   >
+//   <
+//   /
+// >;
```
Comment (ematipico) [inline crates/rslint_parser/src/syntax/jsx/mod.rs:122]: Let's expand the tests with more children: other fragments, other jsx elements, expressions, etc.

**Review by ematipico — CHANGES_REQUESTED** [review]

File: crates/rslint_parser/src/syntax/jsx/mod.rs (hunk 1 above)
Comment (MichaReiser) [reply crates/rslint_parser/src/syntax/jsx/mod.rs:185]: I agree, but it's how the JSX Element parsing works at the moment ([discussion](https://github.com/rome/tools/pull/2224#discussion_r826656207)). This is tracked as part of #2153

File: xtask/codegen/js.ungram (hunk 2 above)
Comment (MichaReiser) [reply xtask/codegen/js.ungram:2186]: This is correct but parsing anything else other than `JsxText` isn't implemented yet and @xunilrj is working on it in #2229.

File: crates/rslint_parser/src/syntax/jsx/mod.rs (hunk 3 above)
Comment (MichaReiser) [reply crates/rslint_parser/src/syntax/jsx/mod.rs:122]: More than happy to do so once other children elements are implemented. But that's all we support at the moment.

File: xtask/codegen/js.ungram (hunk 2 above)
Comment (ematipico) [reply xtask/codegen/js.ungram:2186]: Sorry, I wasn't aware of it

**Review by ematipico — APPROVED** [review]

File: xtask/codegen/js.ungram (hunk 2 above)
Comment (MichaReiser) [reply xtask/codegen/js.ungram:2186]: No worries, I should have mentioned it in the PR description.

**Pull request merged**